### PR TITLE
feat: Add interactive file generation for terraform, helmfile, and packer

### PR DESCRIPTION
**Pull request by osterman**
## Summary

Adds comprehensive support for `atmos terraform generate files` with auto-generation enabled during component execution for all provisioners (terraform, helmfile, packer).

### Core Features
- **Auto-generate files BEFORE path validation** when `auto_generate_files` is enabled, allowing components to be JIT (Just-In-Time) created from stack configuration
- **Interactive prompting** for component and stack selection (like `terraform backend` command)
- **Idempotent file generation** - compares existing content before writing to avoid unnecessary updates
- **Deterministic output** - uses sorted map keys to ensure reproducible file generation across runs
- **Enhanced UI** - shows individual files changed plus a summary line with counts

### Files Modified
- `pkg/terraform/generate/file_generator.go` - Core idempotent generation logic with sorted keys and summary output
- `pkg/terraform/generate/generate.go` - Directory creation before file generation
- `cmd/terraform/generate/files.go` - Interactive prompting for component/stack selection
- `internal/exec/terraform.go` - Auto-generation during terraform execution
- `internal/exec/helmfile.go` - Auto-generation during helmfile execution  
- `internal/exec/packer.go` - Auto-generation during packer execution
- `pkg/schema/schema.go` - Config support for helmfile and packer auto-generation
- `examples/generate-files/` - Complete working example with terraform configurations

## Test Plan

1. Test interactive prompting without arguments:
   ```bash
   atmos -C examples/generate-files terraform generate files
   # Should prompt for component, then stack
   ```

2. Test with component argument only:
   ```bash
   atmos -C examples/generate-files terraform generate files demo
   # Should prompt for stack only
   ```

3. Test with all arguments (no prompting):
   ```bash
   atmos -C examples/generate-files terraform generate files demo -s dev
   # Should generate without prompting
   ```

4. Test idempotency (run twice):
   ```bash
   atmos -C examples/generate-files terraform generate files demo -s dev
   atmos -C examples/generate-files terraform generate files demo -s dev
   # Second run should show "unchanged" files
   ```

5. Test auto-generation during terraform plan:
   ```bash
   cd examples/generate-files
   atmos terraform plan demo -s dev
   # Should auto-generate files before validation
   ```

6. Test helmfile and packer:
   ```bash
   atmos -C examples/generate-files helmfile diff demo -s dev
   atmos -C examples/generate-files packer build demo -s dev
   # Both should auto-generate files if needed
   ```

7. Run test suite:
   ```bash
   make testacc
   ```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Interactive prompts (with shell completion) to select component and stack when generating files; JIT auto-generation for Terraform, Helmfile, and Packer before validation; deterministic HCL/.tfvars output and per-file create/update/skip summaries.

* **Documentation**
  * New blog post, CLI docs, working example, and embedded demos for file-generation workflows.

* **Tests**
  * Broadly expanded tests covering generation, .tfvars, labeled blocks, formatting, templates, and edge cases.

* **Chores**
  * Added example project, README, and ignore/config files for the generate-files demo.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: cmd/terraform/generate/files.go [hunk 1]
```diff
@@ -50,15 +51,41 @@ The generate section in stack configuration supports:
...
+			prompted, err := shared.PromptForStack(cmd, component)
+			if err = shared.HandlePromptError(err, "stack"); err != nil {
+				return err
+			}
+			stack = prompted
+		}
+
+		// Validate stack after prompting (required when component is specified).
+		if component != "" && stack == "" {
+			return errUtils.ErrMissingStack
```
[the remark this answers is not in the document]
Comment (osterman) [reply cmd/terraform/generate/files.go:88]: Fixed in b3209733f - Added `buildConfigAndStacksInfo` helper that properly calls `ParseGlobalFlags` and populates `ConfigAndStacksInfo`, ensuring config-selection flags are respected.

File: examples/generate-files/README.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+│           ├── terraform.tfvars  # Generated
+│           ├── config.json       # Generated
+│           └── README.md         # Generated
+└── stacks/
+    ├── catalog/
+    │   └── demo.yaml             # Component with generate section
+    └── deploy/
+        ├── dev.yaml              # Dev environment
+        └── prod.yaml             # Prod environment
+```
```
[the remark this answers is not in the document]
Comment (osterman) [reply examples/generate-files/README.md]: Fixed in b3209733f - Added `text` language specifier to the code block.

File: internal/exec/helmfile.go [hunk 3]
```diff
@@ -88,6 +89,25 @@ func ExecuteHelmfile(info schema.ConfigAndStacksInfo) error {
...
+			if mkdirErr := os.MkdirAll(componentPath, 0o755); mkdirErr != nil { //nolint:revive
+				return fmt.Errorf("failed to create component directory for auto-generation: %w", mkdirErr)
+			}
+
+			// Generate files before path validation.
+			if genErr := generateFilesForComponent(&atmosConfig, &info, componentPath); genErr != nil {
+				return fmt.Errorf("failed to auto-generate files: %w", genErr)
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (osterman) [reply internal/exec/helmfile.go]: Fixed in b3209733f - Added `&& !info.DryRun` guard to prevent filesystem modifications during dry-run mode.

File: internal/exec/helmfile.go [hunk 4]
```diff
@@ -88,6 +89,25 @@ func ExecuteHelmfile(info schema.ConfigAndStacksInfo) error {
...
+		if generateSection != nil {
+			// Ensure component directory exists for file generation.
+			if mkdirErr := os.MkdirAll(componentPath, 0o755); mkdirErr != nil { //nolint:revive
+				return fmt.Errorf("failed to create component directory for auto-generation: %w", mkdirErr)
+			}
+
+			// Generate files before path validation.
+			if genErr := generateFilesForComponent(&atmosConfig, &info, componentPath); genErr != nil {
+				return fmt.Errorf("failed to auto-generate files: %w", genErr)
+			}
```
[the remark this answers is not in the document]
Comment (osterman) [reply internal/exec/helmfile.go]: Fixed in b3209733f - Added trailing periods to comments and replaced `fmt.Errorf` with `errors.Join` using static sentinels from `errors/errors.go`.

File: internal/exec/packer.go [hunk 5]
```diff
@@ -78,6 +79,25 @@ func ExecutePacker(
...
+		if generateSection != nil {
+			// Ensure component directory exists for file generation.
+			if mkdirErr := os.MkdirAll(componentPath, 0o755); mkdirErr != nil { //nolint:revive
+				return fmt.Errorf("failed to create component directory for auto-generation: %w", mkdirErr)
+			}
+
+			// Generate files before path validation.
+			if genErr := generateFilesForComponent(&atmosConfig, info, componentPath); genErr != nil {
+				return fmt.Errorf("failed to auto-generate files: %w", genErr)
+			}
```
[the remark this answers is not in the document]
Comment (osterman) [reply internal/exec/packer.go]: Fixed in b3209733f - Added trailing periods to comments and replaced `fmt.Errorf` with `errors.Join` using static sentinels from `errors/errors.go`.

File: internal/exec/packer.go [hunk 6]
```diff
@@ -78,6 +79,25 @@ func ExecutePacker(
...
+		}
+	}
```
[the remark this answers is not in the document]
Comment (osterman) [reply internal/exec/packer.go]: Fixed in b3209733f - Added `&& !info.DryRun` guard to prevent filesystem modifications during dry-run mode.

File: internal/exec/terraform.go [hunk 7]
```diff
@@ -209,12 +211,32 @@ func ExecuteTerraform(info schema.ConfigAndStacksInfo) error {
...
+			if mkdirErr := os.MkdirAll(componentPath, 0o755); mkdirErr != nil { //nolint:revive
+				return fmt.Errorf("failed to create component directory for auto-generation: %w", mkdirErr)
+			}
+
+			// Generate files before path validation.
+			if genErr := generateFilesForComponent(&atmosConfig, &info, componentPath); genErr != nil {
+				return fmt.Errorf("failed to auto-generate files: %w", genErr)
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (osterman) [reply internal/exec/terraform.go:238]: Fixed in b3209733f - Added `&& !info.DryRun` guard to prevent filesystem modifications during dry-run mode.

File: internal/exec/terraform.go [hunk 8]
```diff
@@ -209,12 +211,32 @@ func ExecuteTerraform(info schema.ConfigAndStacksInfo) error {
...
+		if generateSection != nil {
+			// Ensure component directory exists for file generation.
+			if mkdirErr := os.MkdirAll(componentPath, 0o755); mkdirErr != nil { //nolint:revive
+				return fmt.Errorf("failed to create component directory for auto-generation: %w", mkdirErr)
+			}
+
+			// Generate files before path validation.
+			if genErr := generateFilesForComponent(&atmosConfig, &info, componentPath); genErr != nil {
+				return fmt.Errorf("failed to auto-generate files: %w", genErr)
+			}
```
[the remark this answers is not in the document]
Comment (osterman) [reply internal/exec/terraform.go]: Fixed in b3209733f - Added trailing periods to comments and replaced `fmt.Errorf` with `errors.Join` using static sentinels from `errors/errors.go`.

File: pkg/terraform/generate/file_generator.go [hunk 9]
```diff
@@ -325,17 +433,128 @@ func serializeToHCL(content map[string]any) ([]byte, error) {
...
+			}
+		case 2:
+			// Double label: resource "type" "name" {}, data "type" "name" {}.
+			// Iterate in sorted order for deterministic output.
+			for _, secondLabel := range sortedKeys(firstMap) {
+				secondValue := firstMap[secondLabel]
+				secondMap, ok := secondValue.(map[string]any)
+				if !ok {
+					return fmt.Errorf("labeled block %s %q %q expects a map, got %T", blockType, firstLabel, secondLabel, secondValue) //nolint:err113
+				}
```
[the remark this answers is not in the document]
Comment (osterman) [reply pkg/terraform/generate/file_generator.go:560]: Fixed in b3209733f - Replaced `fmt.Errorf` with wrapped static error `errUtils.ErrInvalidConfig` from `errors/errors.go`.
